Tests/BasicsTests/TripleTests.swift

```diff
@@ -9,204 +9,244 @@
+                expectedSubArch: nil,
+                expectedVendor: nil,
+                expectedOs: .linux,
+                expectedEnvironment: .android,
+                expectedObjectFormat: .elf
+            ),
+            DataKnownTripleParsing(
+                tripleName: "x86_64-unknown-windows-msvc",
+                expectedArch: .x86_64,
+                expectedSubArch: nil,
+                expectedVendor: nil,
+                expectedOs: .win32,
+                expectedEnvironment: .msvc,
+                expectedObjectFormat: .coff
+            ),
+            DataKnownTripleParsing(
+                tripleName: "wasm32-unknown-wasi",
+                expectedArch: .wasm32,
+                expectedSubArch: nil,
+                expectedVendor: nil,
+                expectedOs: .wasi,
+                expectedEnvironment: nil,
+                expectedObjectFormat: .wasm
+            )
+        ]
+    )
+   func knownTripleParsing(data: DataKnownTripleParsing) {
+            guard let triple = try? Triple(data.tripleName) else {
+                Issue.record("Unknown triple '\(data.tripleName)'.")
+               return
             }
-            XCTAssertEqual(triple.arch, components.arch, file: file, line: line)
-            XCTAssertEqual(triple.subArch, components.subArch, file: file, line: line)
-            XCTAssertEqual(triple.vendor, components.vendor, file: file, line: line)
-            XCTAssertEqual(triple.os, components.os, file: file, line: line)
-            XCTAssertEqual(triple.environment, components.environment, file: file, line: line)
-            XCTAssertEqual(triple.objectFormat, components.objectFormat, file: file, line: line)
-        }
-        XCTAssertTriple("armv7em-apple-none-eabihf-macho", matches: (.arm, .arm(.v7em), .apple, .noneOS, .eabihf, .macho))
-        XCTAssertTriple("x86_64-apple-macosx", matches: (.x86_64, nil, .apple, .macosx, nil, .macho))
-        XCTAssertTriple("x86_64-unknown-linux-gnu", matches: (.x86_64, nil, nil, .linux, .gnu, .elf))
-        XCTAssertTriple("aarch64-unknown-linux-gnu", matches: (.aarch64, nil, nil, .linux, .gnu, .elf))
-        XCTAssertTriple("aarch64-unknown-linux-android", matches: (.aarch64, nil, nil, .linux, .android, .elf))
-        XCTAssertTriple("x86_64-unknown-windows-msvc", matches: (.x86_64, nil, nil, .win32, .msvc, .coff))
-        XCTAssertTriple("wasm32-unknown-wasi", matches: (.wasm32, nil, nil, .wasi, nil, .wasm))
-    }    
-
-    func testTriple() {
+            #expect(triple.arch == data.expectedArch)
+            #expect(triple.subArch == data.expectedSubArch)
+            #expect(triple.vendor == data.expectedVendor)
+            #expect(triple.os == data.expectedOs)
+            #expect(triple.environment == data.expectedEnvironment)
+            #expect(triple.objectFormat == data.expectedObjectFormat)
+   }
+
+    @Test
+    func tripleValidation() {
         let linux = try? Triple("x86_64-unknown-linux-gnu")
```

Comment:
Make this function `throws` and ditch the `?` and the rest of the test can drop the optional unwrapping.